esphome/components/debug/debug_zephyr.cpp

```diff
@@ -86,6 +87,36 @@ std::string DebugComponent::get_reset_reason_() {
 
 uint32_t DebugComponent::get_free_heap_() { return INT_MAX; }
 
+static void fa_cb(const struct flash_area *fa, void *user_data) {
+#if CONFIG_FLASH_MAP_LABELS
+  const char *fa_label = flash_area_label(fa);
+
+  if (fa_label == NULL) {
```

Comment:
```suggestion
  if (fa_label == nullptr) {
```

ESPHome codebase uses nullptr for C++ null pointers.  We missed the one in esp32 in review.